float8_experimental/dynamic_linear/dynamic_float8_linear.py

```diff
@@ -38,6 +39,24 @@ def backward(ctx, gradY):
             None,
         )
 
+def cast_x_to_float8_e4m3fn_pre_hook(module, args):
+    """
+    Hook to cast the incoming activation to `torch.float8_e4m3fn`
+    """
+    return module.cast_to_float8(args[0])
+
+def cast_dldy_to_float8_e5m2_pre_hook(module, grad_output):
```

Comment:
what I do in this function doesn't seem to affect the gradient received by `torch.mm`, unless I modify `grad_output` inplace